### PR TITLE
Warn when view transitions run on a prefere-reduced-motion device

### PR DESCRIPTION
**Pull request by martrapp**
## Changes

In DEV mode warns in the browser console when view transitions are used on a prefer-reduced-motion device.
A user at discord complained that view transitions wouldn't work at all, and @FredKSchott suggested that this could be the reason. Always fun to use an argument from authority!

***Question: *** I would even remove the restriction for DEV mode as it is really hard to find. What do the approvers think? 

## Testing

Manually tested on Windows with Chrome

## Docs

Basically the warning message copies https://docs.astro.build/en/guides/view-transitions/#prefers-reduced-motion. So docs is more than up-to-date.

### CODE REVIEW

**Comment by OliverSpeir** [comment]
Hmm ignoring in dev might be nice, but I assume there’s at least one dev who actually does want to prefer reduce motion for their own sake. 

Could it be a config option ?

**Comment by martrapp** [comment]
> Hmm ignoring in dev might be nice

Ah @OliverSpeir , I'm afraid, I didn't make myself clear: If someone has set up prefer-reduced-motion, we should definitely respect that. But I suspect that many developers only come across users who have set up their system this way in PROD. I therefore believe that the new warning message could also be useful in PROD.

**Comment by martrapp** [comment]
Might need a different text then: "This page was created with view transitions. Because your device ...." or similar. I'm usually the worst candidate to get English text right :)

**Comment by matthewp** [comment]
A lot of people will dislike logging happening in their frontend application. I think in dev mode it is ok, but wouldn't want it in prod.

**Comment by martrapp** [comment]
> A lot of people will dislike logging happening in their frontend application. I think in dev mode it is ok, but wouldn't want it in prod.

Then fine with me as is.

File: packages/astro/components/ViewTransitions.astro [hunk 1]
```diff
@@ -51,6 +51,9 @@ const { fallback = 'animate' } = Astro.props;
 	}
 
 	if (supportsViewTransitions || getFallback() !== 'none') {
+		if (import.meta.env.DEV && window.matchMedia('(prefers-reduced-motion)').matches) {
+			console.warn(`Astro ViewTransitions: all view transition animations, including fallback animation, are disabled as this device has the prefer-reduced-motion setting enabled.`);
```
Comment (ematipico) [inline packages/astro/components/ViewTransitions.astro:55]:
```suggestion
			console.warn(`[transitions]: all view transition animations, including fallback animation, are disabled as this device has the prefer-reduced-motion setting enabled.`);
```

The format is more in line with our server logging

**Review by ematipico — APPROVED** [review]

**Review by Princesseuh — APPROVED** [review]
